src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs

```diff
@@ -4,17 +4,18 @@
 
 using System.IO;
 using System.Reflection;
+using System.Runtime.InteropServices;
 using Xunit;
 
 namespace System.Diagnostics
 {
     /// <summary>Base class used for all tests that need to spawn a remote process.</summary>
     public abstract class RemoteExecutorTestBase : FileCleanupTestBase
     {
-        /// <summary>The CoreCLR host used to host the test console app.</summary>
-        protected const string HostRunner = "corerun";
         /// <summary>The name of the test console app.</summary>
         protected const string TestConsoleApp = "RemoteExecutorConsoleApp.exe";
+        /// <summary>The CoreCLR host used to host the test console app.</summary>
+        protected static readonly string HostRunner = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "CoreRun.exe" : "corerun";
```

Comment:
We need to be able to know whether to use the host or not (using it for net46 will fail, not using it for coreclr will fail), and the cleanest way I came up with to do that was a simple File.Exists test (https://github.com/dotnet/corefx/pull/7606/files/5ca04165e3d4310adfa85349180e1e623df80da9#diff-3943a5ebac23dbf1d2b625df9232c59eR122), but that requires the exact name, not a subset.  If you have a better way in mind, please let me know.